### DISCUSSION TITLE
Spam Add-On Needs UX Improvements

### DISCUSSION THREAD

**Discussion by mcfarlandonline** [ExpressionEngine/ExpressionEngine#3305] [Feature Requests]
EE v 6.2.5
PHP v 7.3
Spam Add-On v 2.0.0

**Primary Issue:** 

> I have a client that heavily uses the [Spam trap Add-On](https://docs.expressionengine.com/latest/add-ons/spam.html), but they find it very difficult there's no indication which entry a potential Spam comment connects to. It seems like the UX should be consistent with the regular/non-trapped Comments dashboard, where entry titles are displayed alongside their corresponding comments. 
 _(Ideally, I think when managing comments, whether in the Spam comment pop-up or on the approved-comment's pages the entry title should be displayed here, too; but one thing at a time...)_

**Secondary Issue:**
> We upgraded this client from EE 5 to EE 6 hoping to see the above issue remedied, but alas, no dice. And briefly, I thought the "manage" icons weren't working for older Spam-trapped comments, (further down the dashboard), because when I clicked the eyeball icon, nothing happened. Eventually, I figured out that the pop-up boxes for Spam comments are sticky/positioned to the top of the interface. You won't see them, unless you scroll back to the top of the screen. 

<img width="1249" alt="Screen Shot 2022-02-16 at 5 14 12 PM" src="https://user-images.githubusercontent.com/20441129/154386620-ce3a8269-50cb-4060-bb1b-481eb9488a72.png">

**Comment by robsonsobral** [comment]
Have you found how to mark a comment as SPAM? I'm still looking for it.

**Comment by mcfarlandonline** [comment]
It's way down at the bottom...
<img width="436" alt="Screen Shot 2022-02-17 at 10 32 27 AM" src="https://user-images.githubusercontent.com/20441129/154548097-7997ff1b-ba89-4df2-ac51-6f14b5f9f209.png">

**Comment by tsamoudakis** [comment]
I've been using the Spam Add-on a lot lately and in addition to the above improvements, I would like to add one more.

### Mark all as Spam with a single button (kind of like "Empty Trash")
There are days that one of my sites receives A LOT of spam comments and in order for me to remove them I have to select the `Show all spam` option from the dropdown, then select them all and then hit submit on `SPAM, then remove from spam trap`. While this works flawlessly when you have 10-20 entries, it slows down significantly when spam entries go over 100, and when they're in the thousands, either it never completes or my browser crashes.

I was expecting that once you mark a couple of hundred of the same comments as spam, the trap would somehow learn from it and immediately remove them straight away in the future. But I'm guessing this is a completely different feature. Therefore having the ability to at least get rid of many spam comments in a faster way, would be a great improvement.

**Comment by intoeetive** [comment]
We have addressed the bit with the popup being hard to notice in [592240b] (secondary issue) but not the primary issue where there is no way to tell where the comment belongs to.

I think it might make sense to store more data with the spam trap, such as original comment_id and the entry_id of where it belongs. That would also allow us to show information about author, if comment was submitted as guest.

Alternatively, we could iuse AJAX to load the comment data, and that could include all those extras (as we can grab comment_id from the serialized entity that's saved in the trap)

Converting this to Feature Request discussion for now
